source/blender/gpu/intern/gpu_material.c

```diff
@@ -877,18 +882,85 @@ static void shade_light_textures(GPUMaterial *mat, GPULamp *lamp, GPUNodeLink **
 		MTex *mtex = lamp->la->mtex[i];
 
 		if (mtex && mtex->tex->type & TEX_IMAGE && mtex->tex->ima) {
-			mat->dynproperty |= DYN_LAMP_PERSMAT;
+			if (lamp->type == LA_AREA) {
+				mat->dynproperty = DYN_LAMP_VEC | DYN_LAMP_CO | DYN_LAMP_AREAMAT;
+			}
+			else {
+				mat->dynproperty |= DYN_LAMP_PERSMAT;
+			}
 
 			float one = 1.0f;
 			GPUNodeLink *tex_rgb;
 
 			GPU_link(mat, "shade_light_texture",
-			         GPU_builtin(GPU_VIEW_POSITION),
-			         GPU_image(mtex->tex->ima, &mtex->tex->iuser, false), GPU_uniform(mtex->size),
+					 GPU_builtin(GPU_VIEW_POSITION),
+					 GPU_image(mtex->tex->ima, &mtex->tex->iuser, false), GPU_uniform(mtex->size),
 					 GPU_select_uniform(&mtex->lodbias, GPU_DYNAMIC_TEX_LODBIAS, NULL, mat->ma),
-			         GPU_dynamic_uniform((float *)lamp->dynpersmat, GPU_DYNAMIC_LAMP_DYNPERSMAT, lamp->ob),
-			         &tex_rgb);
-			texture_rgb_blend(mat, tex_rgb, *rgb, GPU_uniform(&one), GPU_uniform(&mtex->colfac), mtex->blendtype, rgb);
+					 GPU_dynamic_uniform((float *)lamp->dynpersmat, GPU_DYNAMIC_LAMP_DYNPERSMAT, lamp->ob), &tex_rgb);
+
+			if (lamp->type == LA_AREA) {
+				one = 0.95f;
```

Comment:
don't assign `one` to everything else than 1. This value is like a constant in this function as `zero`. All code after can be affected, i know that is not the case here, but later if we add something after we will spend time to looking on a possible rare bug. In your case you can create a different value e.g `float factor = 0.95` and call `GPU_uniform(&factor)`.

Also can you explain why `texture_rgb_blend` must be different for area lamps ?